### ISSUE TITLE
Feature suggestion: RESTful API for other programs to control TTB

### ISSUE THREAD

**Issue by denosawr** [TranslucentTB/TranslucentTB#132]
Suggestion for feature: implement a background web server on a specified port which exposes a RESTful API which allows for other applications to control TTB (with user permission, of course).

**Comment by ethanhs** [comment]
Meh, I think this may add too much bloat. I think the next release (after V4) could go this route, but then we will want to decouple the current UI and the core part.

**Comment by denosawr** [comment]
It doesn't necessarily have to be a RESTful API (even though that would be easiest to control from within Node.JS or Python), but a programmable interface would be nice. Also, moving priority down to low as there probably isn't much demand for such a feature.

**Comment by sylveon** [comment]
Implementing a named pipe from the hook to a client program would be ideal. This way, we could have diverse client programs, and allow the hook to live without a client running either.

**Comment by denosawr** [comment]
I agree with that. (By the way, this should be implemented after #134, as the suggestion would be best done with the listening being done by a hook.) After such an implementation, should we rewrite the client to make use of the new named pipe API?

**Comment by Illustar0** [comment]
It's been 7 years : )
Is there any progress?
Can't wait to control TTB through API : )

**Comment by Illustar0** [comment]
Hello, I'm currently trying to control TranslucentTB externally to implement an idea related to issue #242. My method involves modifying the configuration file in real-time, which works perfectly for basic changes.

However, I would also like to implement smooth color transition animations. This would require writing to the config file at a very high frequency, which is not feasible due to the limitations of file I/O.

It would be fantastic if TranslucentTB could provide an API that allows for external calls. This would open up many possibilities, including the color animations I'm hoping to create.

Thank you for considering this request! @sylveon

**Comment by sylveon** [comment]
TTB has been a bit on the backburner as of the last couple years, been busy with uni. I have plans to rewrite the app using a different architecture that is more resilient to funny stuff the OS can throw at it (also lets me upgrade to WinUI 3). The new architecture should also make it easier to add stuff like this (though I plan to allow users to script their taskbar's appearance, which could remove the need for an external control API entirely).

**Comment by sylveon** [comment]
If you are up to it, I'd gladly take a look at a PR that could expose the color preview system as a window message, so your app could achieve the transition it wants. Perk being that it doesn't change the actual on-disk config.

Just gotta wire up `TaskbarAttributeWorker::ApplyColorPreview` within `TaskbarAttributeWorker::MessageHandler` (e.g. like `WM_TTBFORCEREFRESHTASKBAR` which was added for RoundedTB).

**Comment by Illustar0** [comment]
It doesn't seem too difficult. I'll give it a try. Thanks for pointing out the specific code locations; that will be very helpful.
